Add properties to settings w/ documentation, c++ loading of filename, and python round-trip test (#3808)
openmc-dev/openmc: docs/source/io_formats/settings.rst

  *Default*: 1

.. _properties_file:

--------------------------------------
``<properties_file>`` Element
--------------------------------------

  The ``properties_file`` element indicates whether temperatures and densities 

>   should be read from the properties file instead of from the XML file. It contains

Reviewer: ```suggestion
  The ``properties_file`` element indicates whether temperatures and densities 
  from a properties file should override the properties present in the XML input. It contains
```

  the following parameters:

    :filepath:

>       The path to the properties file to be used to read temperatures and/or densities

Reviewer: ```suggestion
      The path to the properties file
```

    :temperatures:
       Boolean to indicate whether or not to read cell temperatures from the properties file.

       *Default*: True

    :densities:

>        Boolean to indicate whether or not to read cell or material densities from the properties file.

Reviewer: ```suggestion
       Boolean to indicate whether or not to read cell and material densities from the properties file.
```
